src/ser.rs

```diff
@@ -368,6 +392,19 @@ where
     fn serialize_struct(self, _name: &'static str, _len: usize) -> Result<Self::SerializeStruct> {
         Ok(Compound::from_outer(self.outer))
     }
+
+    fn serialize_tuple_struct(
+        self,
+        name: &'static str,
+        len: usize,
+    ) -> Result<Self::SerializeTupleStruct> {
+        match name {
+            "__hematite_nbt_i8_array__"
+            | "__hematite_nbt_i32_array__"
+            | "__hematite_nbt_i64_array__" => Compound::for_seq(self.outer, len as i32, true),
+            _ => Err(Error::UnrepresentableType(stringify!(tuple_struct))),
```

Comment:
Shouldn't this just be `UnrepresentableType("tuple struct")`?

Reply:
yeah, I basically just 1:1 copied what the macro did before, but I can change that if that's more readable

Reply:
Yes, please change.

Reply:
okay will do